### ISSUE TITLE
[Important] Technical Feedback on ACE-Step 1.5 XL (Music Quality & Expressiveness)

### ISSUE THREAD

**Issue by rikunarita** [ace-step/ACE-Step-1.5#1114]
I have been testing ACE-Step 1.5 extensively (including Turbo, and merged variants), and while the model demonstrates strong performance in structural consistency and lyric alignment, there are several critical limitations from a musical perspective.

---

## 1. Vocal Expressiveness (Major Limitation)

The generated vocals are technically accurate but lack expressive nuance:

* Little to no continuous pitch modulation (e.g., pitch bending, scooping, kobushi-like phrasing)
* Vibrato is overly uniform
* Phrasing feels segmented rather than naturally connected
* Emotional delivery is weak, even in contexts that require intensity

This results in vocals that sound “correct” but not “performed.”

---

## 2. Dynamics & Emotional Contour

The model struggles with dynamic contrast:

* Insufficient difference between verse and chorus intensity
* Lack of gradual build-up toward key sections
* Energy transitions feel abrupt rather than intentional

Overall, the music lacks emotional trajectory.

---

## 3. Drum Arrangement & Transitions

Drum patterns are functional but lack musical intent:

* Drum entry often feels abrupt or mechanically placed
* Fills are simplistic and do not act as effective transitions
* No sense of anticipation before drops or chorus
* Groove feels rigid and quantized

This significantly reduces impact, especially in genre styles where rhythm drives emotion.

---

## 4. Groove & Timing

* Timing is overly grid-aligned
* Lacks human-like micro-timing variation (push/pull feel)
* Rhythmic phrasing does not “breathe”

The result is a sterile rhythmic feel.

---

## 5. Melody & Musicality

* Melodies are scale-correct but often lack memorability
* Weak tension-resolution design
* Phrases feel statistically generated rather than intentionally composed

---

## 6. Observed Strengths

* Excellent lyric alignment (almost no mispronunciation)
* Strong structural consistency
* High controllability compared to other systems

These are significant advantages.

---

## 7. Test Conditions

* Models tested:

  * acestep_v1.5_xl (Turbo)
  * acestep_v1.5_xl_merge_sft_turbo_ta_0.5

* Sampling configurations:

  * Sampler: er_sde
  * Scheduler: beta
  * Steps: 12 (Turbo), 32 (Merged)
  * CFG: 1.0 (Turbo), 1.5(Merged)

---

## 8. Key Insight

There is a clear distinction between:

- “Technically correct music generation”
vs
- “Expressive musical performance”

ACE-Step currently excels at the former but struggles with the latter.

While benchmark results may suggest performance comparable to or exceeding systems like Suno v5, in practical listening experience it often feels closer to or below the level of Suno v4 in terms of expressiveness.

---

### (1) Continuous Pitch Modeling

* Enable smoother pitch transitions for expressive singing
* Better modeling of vibrato, pitch bends, and microtonal variation

### (2) Temporal Dynamics Modeling

* Improve energy flow across sections
* Explicit modeling of build-ups and releases

### (3) Drum & Rhythm Intelligence

* Smarter fill generation
* Context-aware drum entry (e.g., pre-chorus anticipation)
* Humanized timing variation

### (4) Multi-Track Interaction Modeling

* Better coordination between vocals, drums, and harmony
* More “band-like” interaction rather than isolated generation

---

## 10. Final Remarks

ACE-Step1.5-XL is already very strong in controllability and structural reliability.

However, improving expressive performance (especially vocals and rhythm) would significantly elevate it from a “correct generator” to a truly musical system.

This is the key gap compared to current expressive audio-first models.

---

Thank you for your work on this project — it has strong potential.

**Comment by ChuxiJ** [comment]
Thank you for your feedbacks

These are AI generation limitations, and we will definitely improve these.

**Comment by gurubusvoyage** [comment]
Reproduced this on Mac (M4 Max, 36GB unified memory, MPS backend), and I think I found the actual root cause: `shift` gets no per-checkpoint auto-correction, unlike `guidance_scale`. Sharing the isolation + the repo evidence in case it saves someone else the debugging time.

**Environment**: local install, based on v0.1.8 + upstream commits through `aa6ef69` (2026-08-01), plus a couple of small local bugfix patches unrelated to this symptom as far as we can tell (fp16→NaN on the MPS text encoder, an `attention_mask` slow-path fix).

**Root cause**: each DiT checkpoint has its own "correct" `shift` baked into the repo's own training registry (`acestep/training_v2/model_discovery.py`, `_BASE_DEFAULTS`):

```python
_BASE_DEFAULTS = {
    "turbo": {"shift": 3.0, "num_inference_steps": 8},
    "base":  {"shift": 1.0, "num_inference_steps": 50},
    "sft":   {"shift": 1.0, "num_inference_steps": 50},
}
```

Our own test harness hardcoded `shift=1.0` once (originally tuned against `base`/`sft`) and never branched it per checkpoint. Applied to `sft`, that's the checkpoint's own correct value, so nothing was ever wrong there. Applied to `turbo`, it's off-preset (should be `3.0`), and that's what produced the segmentation. Two symptoms, one cause: a value that's correct for one checkpoint family silently mis-applied to another, with no safeguard catching it.

**Symptom repro config**: `acestep-v15-turbo` checkpoint + `thinking=True` (5Hz LM code-planning enabled) + **`shift=1.0`** (off-preset for turbo, see above), `guidance_scale=7.0` (ACE-Step auto-corrects this to 1.0 for turbo internally, confirmed via log — no action needed on our end), seed=42, 60s `task_type=text2music` generations. Confirmed via direct listening on **House** (captioned genre "Electronic — House" in our test harness) — this is the only genre with a confirmed baseline listen at this exact broken config; other genres were only generated after applying one of the fixes below, not at the broken baseline. Phrasing came out segmented/discontinuous, matching what's described in this issue.

**Isolated and ruled out on the House case** (same symptom persisted after each change, still at `shift=1.0`):
- Tiled VAE decode boundary — forced non-tiled decode (larger chunk size than latent length) → still segmented
- Caption phrasing — removed "evolving sections"-style wording from the caption → still segmented
- LM size — tested both 1.7B and 4B LM variants → same class of segmentation symptom persisted on both

**What we tried that touched the symptom — one confirmed fix, two we initially misread as fixes**:
1. **`shift=3.0`** (turbo's own correct preset per `_BASE_DEFAULTS` above, everything else unchanged — `thinking=True`) → segmentation gone. This is the actual fix — turbo restored to its own native schedule. Remaining issue on this config, distinct from segmentation: drums stayed timbrally consistent, but pitched/harmonic instruments drifted in timbre across the clip.
2. Switch checkpoint to `acestep-v15-sft` (non-distilled), `inference_steps=50`, still nominally at `shift=1.0` → segmentation gone. **We initially logged this as a second independent fix — it isn't.** `1.0` is `sft`'s own correct preset (see registry above), so this config was never actually broken; we just hadn't noticed we were comparing a working config against a broken one that happened to share a hardcoded number.
3. `thinking=False` (skip the LM code-planning step entirely, DiT + caption/lyrics only), still at `shift=1.0` (still off-preset for turbo) → segmentation gone. **This one we can't explain.** Unlike (2), turbo really is at the wrong shift here, yet the symptom doesn't show. We don't have a theory for why disabling LM code-planning masks an off-preset DiT sampling parameter, and we're not speculating past what we measured.

Combining (1)+(3) or (2)+(3) did *not* help further — e.g. `sft` + `thinking=False` together sounded worse than either alone.

**Raw facts on the open puzzle, no unified theory** — 3 of 4 corners of a 2×2 (turbo checkpoint only, since this axis is specifically about turbo's own off-preset shift):

| | `shift=1.0` (off-preset for turbo) | `shift=3.0` (turbo's own preset) |
|---|---|---|
| `thinking=True` | **segmented** | clean (timbre drift remains) |
| `thinking=False` | clean | untested |

**Suggested improvement**: `guidance_scale` gets auto-corrected for turbo (`generate_music.py`, "Turbo model detected: overriding guidance_scale X -> 1.0") and the repo already carries the correct per-checkpoint `shift` in `_BASE_DEFAULTS` for training — it just isn't consulted at inference time when a caller passes an explicit `shift`. A similar warning/clamp (or at least defaulting an unset `shift` from `_BASE_DEFAULTS` by checkpoint type) would catch exactly this class of bug, since the failure mode (audible segmentation) is much less obvious to diagnose than a NaN would be — anyone hardcoding `shift` across a checkpoint sweep, like we did, will hit this.

Curious whether this reproduces on other backends (CUDA) / variants (XL, turbo-shift3) — happy to share exact params if useful for narrowing it down further.
